Source/WTF/wtf/glib/GRefPtr.h

```diff
@@ -156,157 +162,88 @@ template <typename T> class GRefPtr {
     // Only used for C API functions returning (transfer none) of objects where the above
     // can be established, e.g. objects stored in a global dictionary.
     T* /* (transfer none) */ getUncheckedLifetime() const { return m_ptr; }
-    T& operator*() const { return *m_ptr; }
     ALWAYS_INLINE T* operator->() const { return m_ptr; }
 
     bool operator!() const { return !m_ptr; }
+    explicit operator bool() const { return !!m_ptr; }
 
     // This conversion operator allows implicit conversion to bool but not to other integer types.
     typedef T* GRefPtr::*UnspecifiedBoolType;
     operator UnspecifiedBoolType() const { return m_ptr ? &GRefPtr::m_ptr : 0; }
 
-    GRefPtr& operator=(const GRefPtr&);
-    GRefPtr& operator=(GRefPtr&&);
-    GRefPtr& operator=(T*);
-    template <typename U> GRefPtr& operator=(const GRefPtr<U>&);
+    GRefPtr& operator=(const GRefPtr& other)
+    {
+        GRefPtr ptr = other;
+        swap(ptr);
+        return *this;
+    }
 
-    void swap(GRefPtr&);
-    friend GRefPtr adoptGRef<T>(T*);
+    GRefPtr& operator=(GRefPtr&& other)
+    {
+        GRefPtr ptr = WTFMove(other);
+        swap(ptr);
+        return *this;
+    }
 
-private:
-    static T* hashTableDeletedValue() { return reinterpret_cast<T*>(-1); }
-    // Adopting constructor.
-    GRefPtr(T* ptr, GRefPtrAdoptType) : m_ptr(ptr) {}
+    GRefPtr& operator=(T* other)
+    {
+        GRefPtr ptr = other;
+        swap(ptr);
+        return *this;
+    }
 
-    T* m_ptr;
-};
+    GRefPtr& operator=(std::nullptr_t)
+    {
+        clear();
+        return *this;
+    }
 
-template <typename T> inline GRefPtr<T>& GRefPtr<T>::operator=(const GRefPtr<T>& o)
-{
-    T* optr = o.get();
-    if (optr)
-        refGPtr(optr);
-    T* ptr = m_ptr;
-    m_ptr = optr;
-    if (ptr)
-        derefGPtr(ptr);
-    return *this;
-}
+    void swap(GRefPtr& other)
+    {
+        std::swap(m_ptr, other.m_ptr);
+    }
 
-template <typename T> inline GRefPtr<T>& GRefPtr<T>::operator=(GRefPtr<T>&& o)
-{
-    GRefPtr ptr = WTFMove(o);
-    swap(ptr);
-    return *this;
-}
+private:
+    static T* hashTableDeletedValue() { return reinterpret_cast<T*>(-1); }
 
-template <typename T> inline GRefPtr<T>& GRefPtr<T>::operator=(T* optr)
-{
-    T* ptr = m_ptr;
-    if (optr)
-        refGPtr(optr);
-    m_ptr = optr;
-    if (ptr)
-        derefGPtr(ptr);
-    return *this;
-}
+    friend GRefPtr adoptGRef<T, RefDerefTraits>(T*);
+    enum AdoptTag { Adopt };
+    GRefPtr(T* ptr, AdoptTag) : m_ptr(ptr) { }
```

Comment:
Style checker wants you to write:

```suggestion
    GRefPtr(T* ptr, AdoptTag)
        : m_ptr(ptr) { }
```

But I think I prefer your way, so feel free to ignore it.